src/udev/scsi_id/scsi_id.c

```diff
@@ -198,35 +157,37 @@ static int get_file_options(const char *vendor, const char *model,
                 if (*buf == '#')
                         continue;
 
-                str1 = strsep(&buf, "=");
-                if (str1 && strcaseeq(str1, "VENDOR")) {
-                        str1 = get_value(&buf);
-                        if (!str1) {
-                                retval = log_oom();
+                r = extract_many_words(&buf, "=\",\n", 0, &key, &value, NULL);
+                if (r < 2) {
+                        log_error("Error parsing config file line %d '%s'", lineno, buffer);
+                        retval = -1;
```

Comment:
Hmm, if you are touching this code, can we please make the return values follow the normal style. Especially in this case, where we were returning -ENOMEM, but now we return -EPERM ;]
I had a cursory look and it seems that the details of the return value are ignored, so doing this should make the code nicer without too much refactoring.